src/nvim/api/ui.c

```diff
@@ -112,6 +114,14 @@ void nvim_ui_attach(uint64_t channel_id, Integer width, Integer height,
 
   pmap_put(uint64_t)(connected_uis, channel_id, ui);
   ui_attach_impl(ui);
+
+  if (channel_id <= LONG_MAX) {
```

Comment:
this seems like an anti-pattern. Better to then add assert at channel/job creation that the number is not larger than VARNUMMER_MAX or what it is called.

Reply:
Yep, this "should never" happen, and if it does we want a loud notice.